### ISSUE TITLE
NaN Values in Training and Validation Loss for Large Subvolume Dataset

### ISSUE THREAD

**Issue by riddhisera** [MIC-DKFZ/nnUNet#2002]
Hello nnU-Net Team,

I've been using nnU-Net for segmentation tasks on a dataset of 40 CT scans of coronary arteries. The model performed well on this initial dataset. However, I encountered an issue when expanding my work to a larger set of subvolumes.

**Issue Description:**
After extracting subvolumes from the original dataset, I ended up with approximately 70,000 subvolume files. The intention was to train nnU-Net on these 70k files. The dataset successfully passed the preprocessing validation, but I've been encountering NaN values in both training and validation losses during training, as indicated in the following log output:

```
2024-03-10 17:57:12.983472: Epoch 0 
2024-03-10 17:57:12.984014: Current learning rate: 0.01 
2024-03-10 18:03:27.178388: train_loss nan 
2024-03-10 18:03:27.198700: val_loss nan 
2024-03-10 18:03:27.199236: Pseudo dice [0.245] 
2024-03-10 18:03:27.199863: Epoch time: 374.2 s 
2024-03-10 18:03:27.200424: Yayy! New best EMA pseudo Dice: 0.245 
```

The pseudo dice values seem to vary randomly without showing improvement, and both `train_loss` and `val_loss` consistently show `nan`.

**Additional Observations:**
- Training on a smaller subset (e.g., the first 100 subvolumes) runs successfully without issues.
- Attempting to train on the first 200 subvolumes resulted in the same NaN issue for me, I kept reducing and trying to train my dataset till 182 and found that it works on the first 182 files but not on the first 183 files. When I have the first 181 files and the 183rd file - it works.
- I also tried randomly sampling 100 dataset files from the original 70k, and it does not work. 
- We've verified that there are no NaN values in the dataset itself.

We're open to sharing the dataset with you for closer examination if that would aid in resolving this issue. Your guidance and insights on this matter would be highly valuable to us.

Thank you for your support!
Riddhi

**Comment by riddhisera** [comment]
Update: Training works with nnUNetv2 version 2.1 but not with 2.2

**Comment by Karol-G** [comment]
Dear Riddhi,

Thank you for your detailed message and for using nnU-Net for your segmentation tasks.

To better understand and address this issue, could you please elaborate on the specific reasons behind generating subvolumes from your dataset? Understanding your rationale for this approach will help us provide more targeted advice.

Additionally, could you provide more specific information about the task at hand and the problem you encountered? Details on the exact nature of your segmentation task and any particular challenges you are facing with these subvolumes would be helpful.

For a more thorough investigation, we request the following:
- The dataset.json file, which will give us a better overview of how your dataset is structured.
- The exact preprocessing and train commands you used. This will help us replicate your workflow and identify potential issues.
- The training log and the progress.png of every fold, all within a zip archive. These files are crucial for us to analyze the training progression and pinpoint when and how the NaN values are occurring.

Lastly, it would be immensely helpful if you could send file 182 to karol.gotkowski@dkfz.de. This particular file seems to be pivotal to the issue, and examining it directly could provide significant insights.

We appreciate your cooperation and are committed to resolving this issue. Looking forward to your response.

Best regards,
Karol Gotkowski
